dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowPermissionsHelper.java

```diff
@@ -0,0 +1,298 @@
+     */
+    @CloseDBIfOpened
+    List<WorkflowAction> filterActions(final List<WorkflowAction> actions,
+            final User user, final boolean respectFrontEndRoles,
+            final Permissionable permissionable) throws DotDataException {
+
+        if ((user != null) && roleAPI.doesUserHaveRole(user, cmsAdminRole)) {
+            return actions;
+        }
+
+        List<WorkflowAction> permissionables = new ArrayList<>(actions);
+        if (permissionables.isEmpty()) {
+            return permissionables;
+        }
+
+        final Iterator<WorkflowAction> actionsIterator = permissionables.iterator();
+        while (actionsIterator.hasNext()) {
+            final WorkflowAction action = actionsIterator.next();
+            boolean hasPermission = false;
+            if (null != permissionable) {
+                // Validate if the action has one of the workflow special roles
+                hasPermission = hasSpecialWorkflowPermission(user, respectFrontEndRoles,
+                        permissionable, action);
+            }
+            // Validate if has other rolers permissions
+            if (permissionAPI.doesUserHavePermission(action, PermissionAPI.PERMISSION_USE, user,
+                    respectFrontEndRoles)) {
+                hasPermission = true;
+            }
+            if (!hasPermission) {
+                actionsIterator.remove();
+            }
+        }
+
+        return permissionables;
+    }
+
+
+    /**
+     * Filters the list of actions to display according to the role passed
+     *
+     * @param actions the actions to filter by role
+     * @param role the role selected
+     * @return the list of actions once the filter operation has taken place
+     */
+    @CloseDBIfOpened
+    List<WorkflowAction> filterActions(final List<WorkflowAction> actions, final Role role,
+            final Permissionable permissionable) throws DotDataException {
+
+        final List<WorkflowAction> permissionables = new ArrayList<>(actions);
+        if (permissionables.isEmpty()) {
+            return permissionables;
+        }
+
+        //First try to determine if we're dealing with a role directly mapped to a user.
+        try {
+            final User user = userAPI.loadUserById(role.getRoleKey());
+            //if we're performing a filter on a role that is mapped to a user. Lets say Chris Publisher
+            //for that reason we have a user, with assigned roles etc..
+            return filterActions(permissionables, user, false, permissionable);
+        } catch (Exception nsu) {
+            Logger.debug(this, () -> "Unable to determine role belongs to a user.");
+        }
+
+        final Iterator<WorkflowAction> workflowActionIterator = permissionables.iterator();
+
+        final List<Role> specialRolesHierarchy = getSpecialRolesByPrecedence();
+        final int index = specialRolesHierarchy.indexOf(role);
+        //Determine if we're dealing with a special role
+        if (index >= 0) {
```

Comment:
isn't the same to do specialRolesHierarchy.contains(role)

Reply:
nvm I see you are using the index later

Reply:
yeah.. the index is used to extract a sub-list of roles. Another reason why a list comes handy here instead of a set. =)